pyintegration/deephaven2/dtypes.py

```diff
@@ -99,31 +108,236 @@ def array_from(self, values: Iterable):
+
+    def __init__(self, j_name: str, qst_type: Any, np_dtypes: Tuple, null_value: Any):
+        super().__init__(j_name=j_name, qst_type=qst_type, is_primitive=True)
+        self.np_types = np_dtypes
+        self.null_value = null_value
+
+    def array_of(self, seq):
+        """ Creates a Java char array populated with values from a sequence.
+
+        Args:
+            seq: a sequence of compatible data, e.g. list, tuple, numpy array, Pandas series, etc.
+
+        Returns:
+            a Java array
+
+        Raises:
+            DHError
+        """
+
+        def to_char(el):
+            if el is None:
+                return NULL_CHAR
+            if isinstance(el, int):
+                return el
+            if isinstance(el, str):
+                if len(el) < 1:
+                    return NULL_CHAR
+                return ord(el[0])
+            try:
+                return int(el)
+            except ValueError:
+                return NULL_CHAR
+
+        try:
+            if isinstance(seq, str):
+                return super().array_of([ord(c) for c in seq])
+            elif isinstance(seq, np.ndarray):
+                if seq.dtype == np.uint16:
+                    return super().array_of(seq)
+                elif np.issubdtype(seq.dtype, np.integer):
+                    return super().array_of(seq.astype(np.uint16))
+                elif seq.dtype == np.dtype('U1') and seq.dtype.name in ['unicode32', 'str32', 'string32',
+                                                                        'bytes32']:
+                    arr = np.copy(seq)
+                    arr.dtype = np.uint32
+                    return super().array_of(arr.astype(np.uint16))
+                elif seq.dtype == np.dtype('S1') and seq.dtype.name in ['str8', 'string8', 'bytes8']:
+                    arr = np.copy(seq)
+                    arr.dtype = np.uint8
+                    return super().array_of(arr.astype(np.uint16))
+                elif seq.dtype == object:
+                    return super().array_of(np.array([to_char(el) for el in seq], dtype=np.uint16))
+                else:
+                    # do our best
+                    raise ValueError(
+                        f"Passed in a numpy array, expect integer dtype or one char string dtype, and got {seq.dtype}")
+            elif isinstance(seq, pd.Series):
+                return self.array_of(seq.values)
+            else:
+                return self.array_of(np.asarray(seq))
+        except Exception as e:
+            raise DHError(e, f"failed to create a Java {self.j_name} array.") from e
+
+
+# region convenience enum and module level type aliases
+class DTypes(Enum):
+    """ An enum that defines a predefined set of supported data types in Deephaven. """
+
+    bool_ = DType(j_name="java.lang.Boolean", qst_type=_JQstType.booleanType(), )
+    byte = IntegerDType(j_name="byte", qst_type=_JQstType.byteType(), np_dtypes=(np.int8, np.uint8),
```

Comment:
Should `np.unit8` be included here, since the range doesn't match?  Same comment for other types below.